langchain4j-vertex-ai/src/main/java/dev/langchain4j/model/vertexai/VertexAiEmbeddingModel.java

```diff
@@ -113,6 +144,83 @@ public Response<List<Embedding>> embedAll(List<TextSegment> segments) {
         }
     }
 
+    /**
+     * Calculates the number of tokens for each segment in the input list.
+     *
+     * @param segments a list of TextSegments
+     *
+     * @return a list of tokens counts for each segment
+     */
+    public List<Integer> getTokensCounts(List<TextSegment> segments) {
```

Comment:
minor: maybe `calculateTokenCount`? as opposed to `estimateTokenCount` which we have in other places?

"get" sounds like a getter and It can be unobvious for the user that "get" method does a network call.

Reply:
Good catch again, it's not a getter.
And I hadn't noticed the `TokenCountEstimator` interface.
I'll implement that interface.

Reply:
Not sure it should be `TokenCountEstimator ` as estimator is named like this to emphasize estimation, not the exact count.
At least this is the case for OpenAI as we do not know exactly how they count tokens in some cases, like with tools, thus "estimate".
Also the interface probably dos not fit your use case?